Source/WebKit/GPUProcess/graphics/ScopedRenderingResourcesRequestCocoa.mm

```diff
@@ -60,11 +60,14 @@
     BEGIN_BLOCK_OBJC_EXCEPTIONS
 #if PLATFORM(MAC)
     auto devices = adoptNS(MTLCopyAllDevices());
-    for (id <MTLDevice> device : devices.get())
-        [(_MTLDevice *)device _purgeDevice];
+    for (id<MTLDevice> device : devices.get()) {
+        if ([device respondsToSelector:@selector(_purgeDevice)])
+            [(_MTLDevice *)device _purgeDevice];
```

Comment:
I thought that Objective-C objects would ignore messages they didn't understand?

Reply:
They generate a `unrecognized selector sent to instance` exception. If the Objective-C object was nil it will result in a no-op.